### PR TITLE
DAL: DB entities and Repositories

### PR DESCRIPTION
**Pull request by Thulrig**
Added:
- Entities;
- Interfaces;
- Repositories;
- Context.

I’ve got some questions about UnitOfWork.

### CODE REVIEW

File: StoreAPI/Controllers/OrdersController.cs [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+using Microsoft.EntityFrameworkCore;
+using StoreAPI.Models;
+
+namespace StoreAPI.Controllers
+{
+    [Produces("application/json")]
+    [Route("api/Orders")]
+    public class OrdersController : Controller
+    {
+        private readonly StoreDbContext _context;
```
Comment (turbosega) [inline StoreAPI/Controllers/OrdersController.cs:14]: Controller should not know anything about DBContext!!! 
It should work only with Repositories/Services through interfaces!

File: StoreAPI/Models/Order.cs [hunk 2]
```diff
@@ -0,0 +1,12 @@
+﻿using System;
+
+namespace StoreAPI.Models
+{
+    public class Order
+    {
+        public int OrderId { get; set; }
+        public DateTime OrderDate { get; set; }
+
+        public Product Product { get; set; }
```
Comment (turbosega) [inline StoreAPI/Models/Order.cs:10]: Order entity should not contain any info about products. Only OrderProduct knows about order.

File: StoreAPI/Models/OrderProduct.cs [hunk 3]
```diff
@@ -0,0 +1,16 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace StoreAPI.Models
+{
+    public class OrderProduct
```
Comment (turbosega) [inline StoreAPI/Models/OrderProduct.cs:8]: OrderProduct entity should contain OrderId, and info about Product (Id, quantity). Address should not be here. Address must be in Order entity.

File: StoreAPI/Models/Product.cs [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+    public class Product
+    {
+        [Key]
+        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
+        public int ProductId { get; set; }
+        public string ProductName { get; set; }
+        public int ProductQuantity { get; set; }
+        public string ProductDescription { get; set; }
+
+        public Order Order { get; set; }
```
Comment (turbosega) [inline StoreAPI/Models/Product.cs:19]: Order should not be in Product entity.

File: StoreAPI/Models/Product.cs [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+using System.ComponentModel.DataAnnotations.Schema;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace StoreAPI.Models
+{
+    public class Product
+    {
+        [Key]
+        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
```
Comment (turbosega) [inline StoreAPI/Models/Product.cs:13]: [DatabaseGenerated] attribute is redundant here due to [Key] attribute presence.

File: StoreAPI/Pages/About.cshtml.cs [hunk 6]
```diff
@@ -0,0 +1,18 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc.RazorPages;
+
+namespace StoreAPI.Pages
+{
+    public class AboutModel : PageModel
```
Comment (turbosega) [inline StoreAPI/Pages/About.cshtml.cs:9]: We do not implement any UI parts, so Razor Pages code and views should not be in the API project.

**Review by turbosega — COMMENTED** [review]
Folders bin, obj, Pages and wwwroot should not be in API project and repository!!!

**Comment by Thulrig** [comment]
You've reviewed the old one dal that was wrong  So i should commit #12 or
#13 issue for review.

**Pull request closed without merging**